server/src/main/java/org/elasticsearch/index/query/CoordinatorRewriteContextProvider.java

```diff
@@ -49,18 +48,25 @@ public CoordinatorRewriteContext getCoordinatorRewriteContext(Index index) {
         if (indexMetadata == null) {
             return null;
         }
-        DateFieldMapper.DateFieldType dateFieldType = mappingSupplier.apply(index);
-        if (dateFieldType == null) {
+        CachedTimestampFieldInfo timestampsFieldInfo = mappingSupplier.apply(index);
+        if (timestampsFieldInfo == null) {
             return null;
         }
-        IndexLongFieldRange timestampRange = indexMetadata.getTimestampRange();
-        if (timestampRange.containsAllShardRanges() == false) {
-            timestampRange = indexMetadata.getTimeSeriesTimestampRange(dateFieldType);
-            if (timestampRange == null) {
+
+        // ensure the cached info has the latest ranges from cluster state
+        timestampsFieldInfo.setTimestampRange(indexMetadata.getTimestampRange());
+        timestampsFieldInfo.setEventIngestedRange(indexMetadata.getEventIngestedRange());
```

Comment:
I left a question below as well on this new class. I wonder if these set calls are a bit of a smell. Maybe it feels odd that the mapping supplier returns a timestamp field info object, that needs updating. Is that a signal that maybe we should keep field types and ranges separate, perhaps have a compound field type abstraction and a compound field range abstraction, similar to what we had before but supporting more than a single field?

Reply:
Changed. See above comment.